SeamCarver.js

```diff
@@ -203,13 +203,38 @@ class SeamCarver {
     }
 
     /**
+     * Removes vertical seam.
+     * Recalculates pixels depending on removed pixel.
+     *
+     */
+    removeVerticalSeam(vseam) {
+        for (var row = 0; row < vseam.length; row ++) {
+            var deletedCol = vseam[row];
+
+            // Can ignore last column as we will delete it
+            for (var col = 0; col < this.width - 1; col ++) {
+                if (col >= deletedCol) {
+                    this.energy_matrix[col][row] = this.energy_matrix[col + 1][row];
```

Comment:
It does seem to me too that this is where we will need improving. For now, couldn't you use splice? In the off chance its implementation on by the JS runtime is a little better...